### PR TITLE
Add GHES Manage API client

### PR DESCRIPTION
**Pull request by manue1**
Adds `octokit.rb` support for [GHES Manage API](https://docs.github.com/enterprise-server/rest/enterprise-admin/manage-ghes) endpoints.

Given that the GHES Manage API going forward will be the only supported API for managing GHES instances we have added support for endpoints that up until now were part of the Management Console API (see its deprecation notice [here](https://docs.github.com/en/enterprise-server/rest/enterprise-admin/management-console?apiVersion=2022-11-28#deprecation-of-the-management-console-endpoints)).

----

### Before the change?

* No GHES Manage API support was in place

### After the change?

* All endpoints that were previously covered by the Management Console API client are now also included in the GHES Manage API client

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: lib/octokit/default.rb [hunk 1]
```diff
@@ -12,6 +12,11 @@
   rescue LoadError
     Octokit::Warnable.octokit_warn 'To use retry middleware with Faraday v2.0+, install `faraday-retry` gem'
   end
+  begin
+    require 'faraday/multipart'
+  rescue LoadError
+    Octokit::Warnable.octokit_warn 'To use retry middleware with Faraday v2.0+, install `faraday-multipart` gem'
```
Comment (nickfloyd) [inline lib/octokit/default.rb:18]:
```suggestion
    Octokit::Warnable.octokit_warn 'To use multipart middleware with Faraday v2.0+, install `faraday-multipart` gem; note: this is used by the ManageGHES client for uploading licenses'
```

**Review by nickfloyd — CHANGES_REQUESTED** [review]
Hey @manue1 thanks for the changes here and apologies for the delay on getting a review out to you. ❤️  These changes look good overall, there is one change to the console message produced from the Faraday check - have a look at what I am proposing and let me know what you think.

**Comment by alejndr0** [comment]
> Hey @manue1 thanks for the changes here and apologies for the delay on getting a review out to you. ❤️ These changes look good overall, there is one change to the console message produced from the Faraday check - have a look at what I am proposing and let me know what you think.

Hey nick! Please do not merge this PR yet, just been notified I need to make a quick modification, but I will not be able to do it until Friday :)

**Comment by nickfloyd** [comment]
@alejndr0 Thx for the updates... I am looking at this now and will get it reviewed and hopefully released this morning.

**Review by nickfloyd — APPROVED** [review]

**Pull request merged**